### PR TITLE
(For David) Add support for rescore_user and get_user_score

### PR DESCRIPTION
**Pull request by atpaino**
@garylee1 

**Purpose**
Adds support for the new GET|Post /v205/users/{userId}/score APIs to this client library.

**Testing Plan**
1. Added new unit tests
2. Tested manually on the Sift test account.

### CODE REVIEW

File: sift/client.py [hunk 1]
```diff
@@ -213,6 +213,91 @@ def score(self, user_id, timeout=None, abuse_types=None, version=None):
...
+            user_id:  A user's id. This id should be the same as the user_id used in
+                event calls.
+
+            timeout(optional): Use a custom timeout (in seconds) for this call.
+
+            abuse_types(optional): List of abuse types, specifying for which abuse types a score
+                 should be returned (if scores were requested).  If not specified, a score will
+                 be returned for every abuse_type to which you are subscribed.
+
+            version(optional): Use a different version of the Sift Science API for this call.
```
Comment (ehrmann) [inline sift/client.py]: Is this unimplemented?

File: tests/test_client.py [hunk 2]
```diff
@@ -265,6 +301,58 @@ def test_score_with_timeout_param_ok(self):
...
+        """
+        test_timeout = 5
+        mock_response = mock.Mock()
+        mock_response.content = USER_SCORE_RESPONSE_JSON
+        mock_response.json.return_value = json.loads(mock_response.content)
+        mock_response.status_code = 200
+        mock_response.headers = response_with_data_header()
+        with mock.patch('requests.get') as mock_get:
+            mock_get.return_value = mock_response
+            response = self.sift_client.get_user_score('12345', test_timeout)
```
Comment (ehrmann) [inline tests/test_client.py]: Optional: do you also want to test specifying abuse types?

File: sift/client.py (hunk 1 above)
Comment (atpaino) [reply sift/client.py]: Ah, good catch, I'll remove this.

File: tests/test_client.py (hunk 2 above)
Comment (atpaino) [reply tests/test_client.py]: Sure, added more tests.

**Pull request merged**